appcues/src/main/java/com/appcues/action/ActionProcessor.kt

```diff
@@ -26,9 +26,21 @@ internal class ActionProcessor(override val scope: Scope) : KoinScopeComponent {
         }
     }
 
-    fun process(action: ExperienceAction) {
+    fun process(actions: List<ExperienceAction>) {
         appcuesCoroutineScope.launch {
-            actionQueue.send(action)
+            transformQueue(actions).forEach {
+                actionQueue.send(it)
+            }
+        }
+    }
```

Comment:
this is where we now take in the full list (all of the actions attached to a single button) at once - allows for the new queue transformation to occur.